lib/Dialect/RTL/RTLTypes.cpp

```diff
@@ -88,6 +99,63 @@ static void printRTLElementType(Type element, DialectAsmPrinter &p) {
   p.printType(element);
 }
 
+//===----------------------------------------------------------------------===//
+// Struct Type
+//===----------------------------------------------------------------------===//
+namespace circt {
+namespace rtl {
+bool operator==(const StructType::FieldInfo &a,
+                const StructType::FieldInfo &b) {
+  return a.name == b.name && a.type == b.type;
+}
+llvm::hash_code hash_value(const StructType::FieldInfo &fi) {
+  return llvm::hash_combine(fi.name, fi.type);
+}
+} // namespace rtl
+} // namespace circt
+
+Type StructType::parse(MLIRContext *ctxt, DialectAsmParser &p) {
+  llvm::SmallVector<FieldInfo, 4> parameters;
+  StringRef name;
+  if (p.parseLess())
+    return Type();
```

Comment:
Do these hooks need to emit errors?  It would be good to include a error-producing testcase